src/components/TokenTable.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useCallback } from 'react';
+import { useStaticQuery, graphql } from 'gatsby';
+import { SprkTable, SprkTextInput } from '@sparkdesignsystem/spark-react';
+import debounce from 'lodash/debounce';
+
+const TokenTable = () => {
+  const tokenData = useStaticQuery(
+    graphql`
+      query SassVars {
+        allSassVarsJson {
+          edges {
+            node {
+              description
+              context {
+                name
+                value
+              }
+            }
+          }
+        }
+      }
+    `,
+  );
+  const sassVarData = tokenData.allSassVarsJson.edges.map((item) => ({
+    name: item.node.context.name,
+    value: item.node.context.value,
+    description: item.node.description,
+    flattenedData: [
+      item.node.context.name,
+      item.node.context.value,
+      item.node.description,
+    ]
+      .join(' ')
+      .toLowerCase(),
+  }));
+
+  const [filteredData, setFilteredData] = useState(sassVarData);
+  const [value, setValue] = useState('');
+
+  const debouncedFilterData = useCallback(
+    debounce((searchQuery) => {
+      setFilteredData(
+        sassVarData.filter(({ flattenedData }) => {
+          return flattenedData.includes(searchQuery.toLowerCase());
+        }),
+      );
+    }, 500),
+    [],
+  );
+
+  const handleChange = (event) => {
+    const { value: searchQuery } = event.target;
```

Comment:
i dont think you need this const, also you could set filteredData to be the original data if the query is empty, bypassing the filter